torch/_inductor/ir.py

```diff
@@ -2105,6 +2106,133 @@ def realize(self):
         pass
 
 
+class BufferList(IRNode):
+    def __init__(self, data):
+        super().__init__()
+        self.name = V.graph.register_buffer(self)
+        self.buffers = [
+            ListElemBuffer(layout, self, i) for i, layout in enumerate(data.layouts)
+        ]
+        names = [t.name for t in self.buffers]
+        self.list_name = V.graph.register_list(names)
+        self.data = data
+
+    def __getitem__(self, ind):
+        return TensorBox.create(self.buffers[ind])
+
+    def make_indexer(self, ind):
+        return self.data.layouts[ind].make_indexer()
+
+    def get_name(self):
+        return self.name
+
+    def make_loader(self):
+        def fn():
+            load = self.data.make_loader()
+            return load()
+
+        return fn
+
+    def get_store_function(self):
+        def fn(result):
+            ops.store(self.list_name, self.data.layouts, result)
+
+        return fn
+
+    def decide_layout(self):
+        pass
+
+    def get_device(self):
+        return self.data.layouts[0].device
+
+    def get_layouts(self):
+        return self.data.layouts
+
+    def is_no_op(self):
+        return False
+
+    def get_alias_names(self):
+        return ()
+
+    def get_mutation_names(self):
+        return ()
+
+    def list_arg_count(self):
+        _, (list_reads, list_writes) = self.extract_read_writes()
+        return len(list_reads) + len(list_writes)
+
+    @cache_on_self
+    def normalized_read_writes(self):
+        return self.extract_read_writes()[0]
+
+    @cache_on_self
+    def extract_read_writes(self):
```

Comment:
I'm confused why extract_read_writes live here and not in the dependencies.py

Reply:
Yeah it doesn't have to, I thought the list thing was kind of specific to buffer list I think. I'm fine with moving it to dependencies though and creating an `extract_list_reads_writes` function.